python/lsst/meas/algorithms/detection.py

```diff
@@ -489,6 +515,38 @@ def setEdgeBits(maskedImage, goodBBox, edgeBitmask):
                                                      afwGeom.ExtentI(w, h)), afwImage.LOCAL)
             edgeMask |= edgeBitmask
 
+    @pipeBase.timeMethod
+    def replacePeaks(self, targetFpList, goodPeakFpList):
+        """Replace all of the peaks in one FootprintSet with the overlapping ones from another.
+
+        When this would remove all of the Peaks from a Footprint, we instead remove all but the
+        brightest.
+
+        Parameters
+        ----------
+        targetFpSet : `list`
+            Sequence containing the Footprints whose Peaks we want to replace.
+        goodPeakFpSet : `list`
+            Sequence containing the Footprints whose Peaks we want to use instead.
+        """
+        for targetFp in targetFpList:
+            targetBBox = targetFp.getBBox()
+            newPeaks = afwDet.PeakCatalog(targetFp.getPeaks().getTable())
+            for goodPeakFp in goodPeakFpList:
+                if not targetBBox.overlaps(goodPeakFp.getBBox()):
+                    continue
+                for candidatePeak in goodPeakFp.getPeaks():
+                    if targetFp.contains(candidatePeak.getI()):
+                        newPeaks.append(candidatePeak)
```

Comment:
This is going to break memory contiguity, right?  Can you throw in a `reserve` statement in order to preserve contiguity (maybe put all the matching peaks in a list, then `reserve` and `append` all of them when you have the complete list)?
Ditto on the `extend` later.